firedrake/mg/interface.py

```diff
@@ -18,20 +18,41 @@ def prolong(coarse, fine):
     hierarchy, lvl = utils.get_level(cfs)
     if hierarchy is None:
         raise RuntimeError("Coarse function not from hierarchy")
+    # for skipping hierarchies
+    if hasattr(hierarchy, '_full_hierarchy') == 1:
+        lvl = lvl * hierarchy._skip
+        hierarchy = hierarchy._full_hierarchy
```

Comment:
this is explained above. see 'generalised hierarchies' in firedrakeproject/firedrake-mlmc and our previous discussion on slack channel. I need this functionality for that package.